docs/INSTALL.md

```diff
@@ -0,0 +1,659 @@
+1. [Manual Installation on Linux](#manual-installation-on-linux)
+   1. [Git Clone/Checkout](#git-clonecheckout)
+1. [Dependency Installation](#dependency-installation)
+   1. [Apache Installation/Configuration](#apache-installationconfiguration)
+   1. [PHP Installation/Configuration](#php-installationconfiguration)
+   1. [Database Installation/Configuration](#database-installationconfiguration)
+      1. [MySQL Perfomance tuning \(OPTIONAL\)](#mysql-perfomance-tuning-optional)
+   1. [E-Mail Configuration \(OPTIONAL\)](#e-mail-configuration-optional)
+   1. [Install other Depedencies](#install-other-depedencies)
+   1. [Installation Wizard](#installation-wizard)
+   1. [Configure ILIAS Java RPC server \(OPTIONAL\)](#configure-ilias-java-rpc-server-optional)
+1. [Hardening and Security Guidance](#hardening-and-security-guidance)
+   1. [Secure Files](#secure-files)
+      1. [Place data directory outside of the web root](#place-data-directory-outside-of-the-web-root)
+      1. [Secure Installation Files](#secure-installation-files)
+   1. [Use HTTPS](#use-https)
+      1. [Redirect all unencrypted traffic to HTTPS](#redirect-all-unencrypted-traffic-to-https)
+      1. [Enable HTTP Strict Transport Security](#enable-http-strict-transport-security)
+      1. [Proper SSL configuration](#proper-ssl-configuration)
+   1. [Serve security related Headers](#serve-security-related-headers)
+   1. [Report security issues](#report-security-issues)
+1. [Customizing ILIAS](#customizing-ilias)
+   1. [Plugin Repository](#plugin-repository)
+1. [Upgrading ILIAS](#upgrading-ilias)
+   1. [Minor Upgrade](#minor-upgrade)
+   1. [Major Upgrade](#major-upgrade)
+   1. [Information on Updates](#information-on-updates)
+1. [Upgrading Dependencies](#upgrading-dependencies)
+   1. [PHP](#php)
+   1. [MySQL](#mysql)
+   1. [ImageMagick](#imagemagick)
+1. [Contribute](#contribute)
+   1. [Pull Requests](#pull-requests)
+   1. [Reference System](#reference-system)
+
+<!-- /MarkdownTOC -->
+
+<a name="system-requirements"></a>
+# System Requirements
+
+The necessary hardware to run an ILIAS installation is always dependent from the number of users and the kind of usage.
+
+<a name="cpu"></a>
+## CPU
+
+We RECOMMEND a common dual core server CPU.
+
+<a name="memory"></a>
+## Memory
+
+Memory requirements are greatly variable, depending on the number of users and server activity. We RECOMMEND a minimum of 4096 MB.
+
+<a name="harddrive"></a>
+## Harddrive
+
+We RECOMMEND 250 GB for usual ILIAS deployments. 25 GB would be used by the operating system and ILIAS itself. 225 GB would remain for the database and files.
+
+<a name="bandwidth"></a>
+## Bandwidth
+
+We RECOMMEND at least 100 Mbit/sec. for the web server WAN connection.
+
+<a name="recommended-setup-for-running-ilias"></a>
+## Recommended Setup for Running ILIAS
+
+For best results we RECOMMEND:
+
+  * Debian GNU Linux 8 / Red Hat Enterprise Linux 7 / Ubuntu 16.04 LTS
+  * MySQL 5.5 / MariaDB
```

Comment:
I would recommend MySQL >=5.7 as the recommended setup, since 5.5 suffers from multiple known issues:
* Suboptimal InnoDB defaults (such as `innodb_file_per_table` set to `false` by default)
* Suboptimal general defaults (such as `optimizer_search_depth` set way too high by default)
* Bad query optimization in some edge cases (which can cause database lock-ups when using DataCollection)

All these issues were fixed in MySQL 5.7, which is the default version in Ubuntu 16.04 by the way.